lib/practice_exercises.rb

```diff
@@ -1,13 +1,49 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: constant
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  return true if string.empty? || string.length == 1
+
+  # return everything that not match the pattern with an empty space
+  string_modification = string.gsub(/[^a-zA-Z0-9]/, '').downcase
+
+  index = 0
+  while index < string_modification.length / 2
+    if string_modification[index] == string_modification[string_modification.length - index - 1]
+      index += 1
+    else
+      return false
+    end
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n^2)
+# Space Complexity: O(m*n)
+
+
 def longest_prefix(strings)
```

Comment:
Your time complexity is O(nm) and your space O(m) where m is the length of the shortest string and n is the number of strings.